### ISSUE TITLE
Template as Macros

### ISSUE THREAD

**Issue by abelokoj-github** [texstudio-org/texstudio#2678]
### Describe the feature and the current behavior/state

I am not sure if this is possible, But I would like a situation where users will be able to download document templates (with setups such as page margins, styles and other custom user designs) as Macros that can be used to start a new document with different preference.
### Who will benefit with this feature?
I am aware that there are a few templates in TXS already, but I think it will b useful if there are more that can be downloaded as Macros where users can contribute say like Book, Letter, CV, report, thesis   and other useful templates that one can download as a Macro and just click to have that document setup.

This will be especially useful for beginners as well as those intending to create more complex document &  or book designs.

**Comment by sunderme** [comment]
apart from the download part, there are already document templates ...

**Comment by abelokoj-github** [comment]
> So, you want a central repository for document templates like there is one for macros ?

Yes, I believe that will be very useful. What TXS currently has as templates,  That way, users can have predefined templates for stuffs like letters, thesis, Books with already defined margins and additional styles (Just like Overleaf has such templates). But this should be in something like a macro repository where users can just click over to download what they need for the current document without having to go through the stress of starting up  afresh.

**Comment by dbitouze** [comment]
BTW, would Overleaf allow (in terms of license and technically via APIs) that the templates it provides could be accessed directly from TXS (in a submenu)?

**Comment by sunderme** [comment]
the templates seem to be open source, but there seems not to be an api.

**Comment by sunderme** [comment]
I have implemented online repository with https://github.com/texstudio-org/texstudio-template
Now, someone needs to provide some content (or adapt it for another repository)
